OpenQA/Isotovideo/Utils.pm

```diff
@@ -14,13 +14,13 @@ use Try::Tiny;
 our @EXPORT_OK = qw(git_rev_parse checkout_git_repo_and_branch
   checkout_git_refspec handle_generated_assets load_test_schedule);
 
-sub git_rev_parse ($dirname) {
-    return 'UNKNOWN' unless -e "$dirname/.git";
+sub git_rev_parse ($dirname, $cmd_prefix = '') {
     $dirname = path($dirname)->realpath;
-    my $checksafe = q{git config --global --get safe.directory | grep -q};
-    my $addsafe = q{HOME=$(mktemp -d --tmpdir os-autoinst-git.XXXXX) && git config --global --add safe.directory};
-    my $version = qx{($checksafe "$dirname" && git -C "$dirname" rev-parse HEAD || $addsafe "$dirname" && git -C "$dirname" rev-parse HEAD && rm -r \$HOME)};
-    $version ||= '(unreadable git hash)';
+    chomp(my $version = qx{$cmd_prefix git -C "$dirname" rev-parse HEAD 2>&1});
+    return $version if $? == 0;
+    return 'UNKNOWN' unless $version =~ /(git config.*safe.directory.*$)/;
+    my $addsafe = 'TMPDIR=$(mktemp -d --tmpdir os-autoinst-git.XXXXX) && HOME=$TMPDIR && ' . $1;
+    $version = qx{$addsafe && git -C "$dirname" rev-parse HEAD && rm -r \$TMPDIR} || '(unreadable git hash)';
```

Comment:
if this line is not covered then this likely means that https://github.com/os-autoinst/os-autoinst/pull/2121/files#diff-42b9d43b3da224eb09923bc4972451da136f214a1e03b139f17033c85160a54dR31 does not return successfully, i.e. `command -v sudo >/dev/null && sudo -u nobody true` does not work within our CI environment.

Reply:
I suppose marking as uncoverable is good enough if adding (and using) an additional user becomes too complicated.